time-based-packet-transmission-offload.md

```diff
@@ -9,27 +9,121 @@ Because of the time-fidelity requirements of this feature, there will not be sof
 
 ## Table of Context
 
+- [Time based Packet Transmission Offload (TPTO)](#time-based-packet-transmission-offload-tpto)
+  - [Table of Context](#table-of-context)
 - [Winsock](#winsock)
+  - [Checking for TPTO Capability](#checking-for-tpto-capability)
+  - [Enable TPTO](#enable-tpto)
+  - [Sending Packets](#sending-packets)
 - [TCPIP](#tcpip)
+  - [Queue](#queue)
+    - [Unknown](#unknown)
 - [NDIS](#ndis)
 - [Appendix](#appendix)
+  - [Linix](#linix)
 
 # Winsock
 
 The proposed Winsock API for TPTO is as follows.
 
+## Checking for TPTO Capability
+
+An app first checks for TPTO support by querying the `SO_TPTO_SUPPORT` socket option.  
+The option value is a `TPTO_SUPPORT_FLAGS` enum with flags describing the supported capabilities:
+
+```C
+typedef enum _TPTO_SUPPORT_FLAGS {
+    TPTO_SUPPORT_FLAG_NONE          = 0x00,
+    TPTO_SUPPORT_FLAG_SUPPORTED     = 0x01,
+} TPTO_SUPPORT_FLAGS;
+```
+
+## Enable TPTO
+setsockopt with SO_TXTIME
```

Comment:
Instead of two socket options, `SO_TPTO_SUPPORT` and `SO_TXTIME`, let's just have one, `SO_TXTIME`. You query it to see if the feature is supported:
1. If you query it and it returns `NOT_SUPPORTED` the feature isn't supported.
2. If you query it and it returns success, the value of the return output indicates `TRUE` or `FALSE` to indicate if it's currently enabled or not.

Reply:
setsockopt returns 0 or SOCKET_ERROR.
WSAENOPROTOOPT from WSAGetLastError should be good to represent not supporoted.
I will update.